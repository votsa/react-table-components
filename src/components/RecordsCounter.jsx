import React, { PropTypes } from 'react';

const RecordsCounter = (props) => {
  const {
    total,
    currentPage,
    perPage,
  } = props;

  const from = (Number(currentPage) * Number(perPage)) + 1;
  const to = (Number(currentPage) * Number(perPage)) + Number(perPage);

  return (
    <div className="rtc-records-counter">
      {`Showing ${from} to ${to < total ? to : total} of ${total} records`}
    </div>
  );
};

RecordsCounter.defaultProps = {
  perPage: 10,
  currentPage: 0,
};

RecordsCounter.propTypes = {
  perPage: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  currentPage: PropTypes.number,
  total: PropTypes.number.isRequired,
};

export default RecordsCounter;
