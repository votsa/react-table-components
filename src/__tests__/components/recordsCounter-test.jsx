import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import RecordsCounter from '../../components/RecordsCounter';

describe('<RecordsCounter />', () => {
  it('shows records count', () => {
    const shallowRenderer = createRenderer();

    shallowRenderer.render(
      <RecordsCounter
        total={1000}
        currentPage={0}
        perPage={10}
      />,
    );

    const result = shallowRenderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
