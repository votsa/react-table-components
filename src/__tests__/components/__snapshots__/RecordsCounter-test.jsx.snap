// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecordsCounter /> shows records count 1`] = `
<div
  className="rtc-records-counter"
>
  Showing 1 to 10 of 1000 records
</div>
`;
